From 00739d4e137e1a8aa637666ba0c6dbfdb308fcc2 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Wed, 20 Jul 2016 22:56:33 +0200
Subject: [PATCH] vxlan: fix help and man text

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1269528
Upstream Status: iproute2.git commit 5cd64c979f97a

commit 5cd64c979f97ac1590e7bf28ae9b1adbd7673d3a
Author: Thomas Faivre <thomas.faivre@6wind.com>
Date:   Thu Jan 14 18:10:19 2016 +0100

    vxlan: fix help and man text

    Options 'group' and 'remote' cannot take 'any' as value but 'local' can.

    Signed-off-by: Thomas Faivre <thomas.faivre@6wind.com>
    Signed-off-by: Nicolas Dichtel <nicolas.dichtel@6wind.com>
---
 ip/iplink_vxlan.c     | 2 +-
 man/man8/ip-link.8.in | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/ip/iplink_vxlan.c b/ip/iplink_vxlan.c
index 2ca97a4..015d7d9 100644
--- a/ip/iplink_vxlan.c
+++ b/ip/iplink_vxlan.c
@@ -23,7 +23,7 @@
 
 static void print_explain(FILE *f)
 {
-	fprintf(f, "Usage: ... vxlan id VNI [ { group | remote } ADDR ] [ local ADDR ]\n");
+	fprintf(f, "Usage: ... vxlan id VNI [ { group | remote } IP_ADDRESS ] [ local ADDR ]\n");
 	fprintf(f, "                 [ ttl TTL ] [ tos TOS ] [ dev PHYS_DEV ]\n");
 	fprintf(f, "                 [ dstport PORT ] [ srcport MIN MAX ]\n");
 	fprintf(f, "                 [ [no]learning ] [ [no]proxy ] [ [no]rsc ]\n");
diff --git a/man/man8/ip-link.8.in b/man/man8/ip-link.8.in
index b764019..3ee1ad6 100644
--- a/man/man8/ip-link.8.in
+++ b/man/man8/ip-link.8.in
@@ -333,7 +333,8 @@ the following additional arguments are supported:
 .RB " ] [ { " group " | " remote " } "
 .I IPADDR
 .R " ] [ "
-.BI local " IPADDR "
+.B local
+.RI "{ "IPADDR " | "any " } "
 .R " ] [ "
 .BI ttl " TTL "
 .R " ] [ "
-- 
1.8.3.1

